### PR TITLE
add constants in one file

### PR DESCRIPTION
**Pull request by r14chandra**

### CODE REVIEW

File: ros/lib/constants.py [hunk 1]
```diff
@@ -32,3 +27,69 @@ class CloudProvider(Enum):
 class OperatingSystem(Enum):
     """Supported Operating Systems"""
     RHEL = 'RHEL'
+
+
+class SystemStates(Enum):
```
Comment (srdeotarse) [inline ros/lib/constants.py:32]: I can see that values ("Oversized",  "Undersized" etc) of different keys in ``SYSTEM_STATES`` is similar to elements in ``SYSTEM_STATES_EXCEPT_EMPTY``. We can avoid code redundancy by using  ``SystemStatesWithKeys.SYSTEM_STATES.value.values()`` where we are using ``SystemStates.SYSTEM_STATES_EXCEPT_EMPTY.value`` to iterate over values.

File: ros/api/v1/hosts.py [hunk 2]
```diff
@@ -184,12 +170,12 @@ def build_system_filters():
             modified_states = []
             for state in states:
                 state = state.capitalize()
-                if state not in SYSTEM_STATES_EXCEPT_EMPTY:
+                if state not in SystemStates.SYSTEM_STATES_EXCEPT_EMPTY.value:
```
Comment (srdeotarse) [inline ros/api/v1/hosts.py:173]: We can add ``SystemStatesWithKeys.SYSTEM_STATES.value.values()`` here as suggested in ``constants.py`` file

File: ros/api/v1/hosts.py [hunk 3]
```diff
@@ -184,12 +170,12 @@ def build_system_filters():
...
                     abort(400, message='values are not matching')
                 modified_states.append(state)
             filters.append(System.state.in_(modified_states))
         else:
-            filters.append(System.state.in_(SYSTEM_STATES_EXCEPT_EMPTY))
+            filters.append(System.state.in_(SystemStates.SYSTEM_STATES_EXCEPT_EMPTY.value))
```
Comment (srdeotarse) [inline ros/api/v1/hosts.py:178]: Similar here as well.

**Review by srdeotarse — COMMENTED** [review]
@r14chandra Thanks for the PR!

I have added my suggestions related to reducing code redundancy. But, I want to know whether we have any particular reason to have both``SystemStates.SYSTEM_STATES_EXCEPT_EMPTY`` and ``SystemStatesWithKeys.SYSTEM_STATES`` when only the later can be used wherever first one is used?

**Comment by srdeotarse** [comment]
@r14chandra Just reviewed your PR after the update. Thanks for considering suggested changes. Looks good to me ✅

**Comment by r14chandra** [comment]
\retest

File: ros/lib/constants.py [hunk 4]
```diff
@@ -32,3 +27,57 @@ class CloudProvider(Enum):
...
+    SYSTEM_STATES = {
+        "INSTANCE_OVERSIZED": "Oversized",
+        "INSTANCE_UNDERSIZED": "Undersized",
+        "INSTANCE_IDLE": "Idling",
+        "INSTANCE_OPTIMIZED_UNDER_PRESSURE": "Under pressure",
+        "STORAGE_RIGHTSIZING": "Storage rightsizing",
+        "OPTIMIZED": "Optimized",
+        "NO_PCP_DATA": "Waiting for data"
+    }
+    OPTIMIZED_SYSTEM = "OPTIMIZED"
```
Comment (upadhyeammit) [inline ros/lib/constants.py:42]: Do we require dict in Enum? as example,
~~~
>>> class TestEnum(Enum):
...     CAT="TOM"
...     DOG="GOOFY"
... 
>>> TestEnum.CAT
<TestEnum.CAT: 'TOM'>
>>> TestEnum.DOG
<TestEnum.DOG: 'GOOFY'>
>>> TestEnum.__members__.values()
dict_values([<TestEnum.CAT: 'TOM'>, <TestEnum.DOG: 'GOOFY'>])
>>> TestEnum._member_names_
['CAT', 'DOG']
~~~

File: ros/lib/constants.py [hunk 5]
```diff
@@ -32,3 +27,57 @@ class CloudProvider(Enum):
...
+        MEMORY_UNDERSIZED='Memory utilization is too high',
+        MEMORY_UNDERSIZED_BY_PRESSURE='System is suffering from memory pressure',
+        CPU_OVERSIZED='CPU utilization is very low',
+        CPU_UNDERSIZED='CPU utilization is too high',
+        CPU_UNDERSIZED_BY_PRESSURE='System is suffering from CPU pressure',
+        IO_OVERSIZED='I/O utilization is very low',
+        IO_UNDERSIZED='I/O utilization is too high',
+        IO_UNDERSIZED_BY_PRESSURE='System is suffering from IO pressure',
+        IDLE='System is IDLE',
+    )
```
Comment (upadhyeammit) [inline ros/lib/constants.py:58]: Same we don't require dict?

File: ros/lib/constants.py [hunk 6]
```diff
@@ -32,3 +27,57 @@ class CloudProvider(Enum):
...
+
+# recommendations
+NEWLINE_SEPARATOR = '\n'
+RULES_COLUMNS = ['rule_id', 'description', 'reason', 'resolution', 'condition']
+INSTANCE_PRICE_UNIT = 'USD/hour'
+
+# notification
+BUNDLE = "rhel"
+APPLICATION = "resource-optimization"
+EVENT_TYPE = "new-suggestion"
```
Comment (upadhyeammit) [inline ros/lib/constants.py:83]: I understand these are few vars, however we can still go ahead with separate Enum class for these to maintain the consistency?

**Review by upadhyeammit — CHANGES_REQUESTED** [review]

**Comment by r14chandra** [comment]
Thanks @srdeotarse @upadhyeammit  for the review!

**Pull request merged**